// pepipost/lib/Models/Settings.js
/**
 * PepipostLib
 *
 * This file was automatically generated by APIMATIC v2.0 ( https://apimatic.io )
 */

'use strict';

const BaseModel = require('./BaseModel');

/**
 * Creates an instance of Settings
 */
class Settings extends BaseModel {
    /**
     * @constructor
     * @param   {Object}  obj    The object passed to constructor
     */
    constructor(obj) {
        super(obj);
        if (obj === undefined || obj === null) return;
        this.footer = this.constructor.getValue(obj.footer);
        this.clicktrack = this.constructor.getValue(obj.clicktrack);
        this.opentrack = this.constructor.getValue(obj.opentrack);
        this.unsubscribe = this.constructor.getValue(obj.unsubscribe);
        this.bcc = this.constructor.getValue(obj.bcc);
    }

    /**
     * Function containing information about the fields of this model
     * @return   {array}   Array of objects containing information about the fields
     */
    static mappingInfo() {
        return super.mappingInfo().concat([
            { name: 'footer', realName: 'footer' },
            { name: 'clicktrack', realName: 'clicktrack' },
            { name: 'opentrack', realName: 'opentrack' },
            { name: 'unsubscribe', realName: 'unsubscribe' },
            { name: 'bcc', realName: 'bcc' },
        ]);
    }

    /**
     * Function containing information about discriminator values
     * mapped with their corresponding model class names
     *
     * @return   {object}  Object containing Key-Value pairs mapping discriminator
     *                     values with their corresponding model classes
     */
    static discriminatorMap() {
        return {};
    }
}

module.exports = Settings;
